packages/arb-token-bridge-ui/src/components/App/App.tsx

```diff
@@ -291,27 +293,78 @@ function AppContent() {
 // https://github.com/orgs/WalletConnect/discussions/2733
 // https://github.com/wagmi-dev/references/blob/main/packages/connectors/src/walletConnect.ts#L114
 const searchParams = new URLSearchParams(window.location.search)
-const targetChainKey = searchParams.get('walletConnectChain')
+const targetChainKey = searchParams.get('sourceChain')
 
 const { wagmiConfigProps, rainbowKitProviderProps } = getProps(targetChainKey)
 
 // Clear cache for everything related to WalletConnect v2.
 //
 // TODO: Remove this once the fix for the infinite loop / memory leak is identified.
 Object.keys(localStorage).forEach(key => {
-  if (key === 'wagmi.requestedChains' || key.startsWith('wc@2')) {
+  if (
+    key === 'wagmi.requestedChains' ||
+    key === 'wagmi.store' ||
+    key.startsWith('wc@2')
+  ) {
     localStorage.removeItem(key)
   }
 })
 
 function ConnectedChainSyncer() {
+  const { isSmartContractWallet, isLoading: isLoadingAccountType } =
+    useAccountType()
   const [shouldSync, setShouldSync] = useState(false)
   const [didSync, setDidSync] = useState(false)
+  const { disconnect } = useDisconnect({
+    onSettled: onDisconnectHandler
+  })
 
   const [{ sourceChain, destinationChain }, setQueryParams] =
     useArbQueryParams()
   const { chain } = useNetwork()
 
+  const setSourceChainToConnectedChain = useCallback(() => {
+    if (typeof chain === 'undefined') {
+      return
+    }
+
+    const { sourceChainId: sourceChain, destinationChainId: destinationChain } =
+      sanitizeQueryParams({
+        sourceChainId: chain.id,
+        destinationChainId: undefined
+      })
+
+    setQueryParams({ sourceChain, destinationChain })
+  }, [chain, setQueryParams])
+
+  useEffect(() => {
+    if (typeof chain === 'undefined') {
+      return
+    }
+    if (
+      isSmartContractWallet &&
+      !isLoadingAccountType &&
+      sourceChain !== chain.id
+    ) {
+      const chainName = getNetworkName(chain.id)
+
+      setSourceChainToConnectedChain()
+
+      window.alert(
+        `You connected to the app with a smart contract wallet on ${chainName}. In order to properly enable transfers, we will now reload the app.\n\nPlease reconnect after the reload.`
```

Comment:
very nit, but it's not preferred to use `we` or first person here, instead: 

`You're connected to the app with a smart contract wallet on ${chainName}. In order to properly enable transfers, the app will now reload.\n\nPlease reconnect after the reload.`